cumulus/parachains/runtimes/test-utils/src/test_cases.rs

```diff
@@ -91,3 +91,55 @@ pub fn change_storage_constant_by_governance_works<Runtime, StorageConstant, Sto
 			);
 		})
 }
+
+/// Test-case makes sure that `Runtime` can change storage constant via governance-like call
+pub fn set_storage_keys_by_governance_works<Runtime>(
+	collator_session_key: CollatorSessionKeys<Runtime>,
+	runtime_para_id: u32,
+	runtime_call_encode: Box<dyn Fn(frame_system::Call<Runtime>) -> Vec<u8>>,
+	storage_items: Vec<(Vec<u8>, Vec<u8>)>,
+	initialize_storage: impl FnOnce() -> (),
+	assert_storage: impl FnOnce() -> (),
+) where
+	Runtime: frame_system::Config
+		+ pallet_balances::Config
+		+ pallet_session::Config
+		+ pallet_xcm::Config
+		+ parachain_info::Config
+		+ pallet_collator_selection::Config
+		+ cumulus_pallet_parachain_system::Config,
+	ValidatorIdOf<Runtime>: From<AccountIdOf<Runtime>>,
+{
+	let mut runtime = ExtBuilder::<Runtime>::default()
+		.with_collators(collator_session_key.collators())
+		.with_session_keys(collator_session_key.session_keys())
+		.with_para_id(runtime_para_id.into())
+		.with_tracing()
+		.build();
+	runtime.execute_with(|| {
+		initialize_storage();
+	});
+	runtime.execute_with(|| {
+		// encode `kill_storage` call
+		let kill_storage_call = runtime_call_encode(frame_system::Call::<Runtime>::set_storage {
```

Comment:
@claravanstaden small nits, name `kill_storage` here is kind of misleading, because it does `set_storage`.
Also I think that this introduced `set_storage_keys_by_governance_works` is not necessary and its usage can be replaced with four calls of `change_storage_constant_by_governance_works` in-place.

Reply:
Ok cool! Would it be OK if I create an internal ticket for this? Just because the Ethereum Deneb fork is tomorrow.

Reply:
Originally I used kill_storage to reset nonces so this is leftover naming from that and needs to change. 

The issue with `change_storage_constant_by_governance_works` is that it is bound to storage items which implement [Get\<StorageConstantType>](https://github.com/paritytech/polkadot-sdk/pull/3029/files/8bb432a4e8ad52ed4a9b34e81f6ce27a82be81b7#diff-4db90058dd96cd36eaa6d1751e8937386185c594d86dca5f2bdc689554430f97R42). This limits its usage to storage constants that created in `parameter_types!` macro and only tests single keys. This will not work for pallets with a StorageMap. `set_storage_keys_by_governance_works` does this in a more general way by allowing two closures, one to initialize storage before modification, and one to asset storage after modification.

The tests that use `change_storage_constant_by_governance_works` could be re-implemented in terms of `set_storage_keys_by_governance_works` as it is more general, but that would affect 7 runtime tests which use it. Should we rather do that?

Reply:
@alistair-singh yes, you're right, `change_storage_constant_by_governance_works` was dedicated more for `parameter_types! { pub storage..` stuff. Or maybe `change_storage_constant_by_governance_works` can internally just call `set_storage_keys_by_governance_works`, anyway, if we could make it nicer, lets do it, but this is very low prio,

@claravanstaden we don't have access to [your](https://linear.app/snowfork/issue/SNO-840), it is related to the `polkadot-sdk` code directly, so it is also fine to create issue here.